make/Program.mk

```diff
@@ -69,16 +69,26 @@ endif
 
 .PHONY: flash
 flash: all test_softdevice flash_mbr
+ifeq ($(strip $(grep -q Microsoft /proc/version)),)
+	@echo Flashing: $(HEX)
+	nrfjprog.exe -f nrf52 --program $(HEX) --sectorerase
+	nrfjprog.exe -f nrf52 --reset
+else ifeq ($(USE_BOOTLOADER),1)
 	$(Q)printf "r\n" > $(BUILDDIR)flash.jlink
 ifdef ID
 	$(Q)printf "w4 $(ID_FLASH_LOCATION), 0x$(ID_SECON) 0x$(ID_FIRST)\n" >> $(BUILDDIR)flash.jlink
 endif
-ifeq ($(USE_BOOTLOADER),1)
 	$(Q)printf "loadfile $(MERGED_HEX) \nr\ng\nexit\n" >> $(BUILDDIR)flash.jlink
-else
-	$(Q)printf "loadfile $(HEX) \nr\ng\nexit\n" >> $(BUILDDIR)flash.jlink
+	$(Q)$(JLINK) $(JLINK_FLAGS) $(BUILDDIR)flash.jlink
+else ifeq ($(USE_BOOTLOADER),0)
```

Comment:
We also want to leave this line as just an `else` with no check. USE_BOOTLOADER may not be defined at all, in which case it won't be equal to zero.

Reply:
Honestly,  the better design might just be to leave the logic that creates the flash.jlink file alone. It shouldn't hurt anything to create that file on windows even if you don't use it. Then you can create one if check that decides if you're windows and uses nrfjprog or else falls back on the existing JLINK infrastructure

Reply:
fwiw, `ifeq ($(USE_BOOTLOADER),)` will evaluate to true if the variable is undefined